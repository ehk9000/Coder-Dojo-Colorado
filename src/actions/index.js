export const addUser = (user) => ({
  type:'ADD_USER',
  user
});

export const isFetching = (bool) => ({
  type: 'IS_FETCHING',
  isFetching: bool
});