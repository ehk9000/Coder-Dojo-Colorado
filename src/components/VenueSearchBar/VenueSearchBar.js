import React from 'react';
import { connect } from 'react-redux';

function VenueSearchBar() {
  let cityInput;
  

  function handleSubmit(e) {
    e.preventDefault();

    console.log(cityInput)
  }

  return (
    <form 
      className="VenueSearchBar"
      onSubmit={handleSubmit}
    >
      <h3>Find a Dojo to Attend</h3>
      <p>Learn technology in an informal, creative and social environment. Find a dojo near you.</p>
      <label htmlFor="city-input">
        <input 
          type="text" 
          id="city-input"
          onChange={(e) => {
            cityInput = e.target.value;
          }} />
      </label>
      <button type="submit" className="city-search-btn">Find a Dojo</button>
    </form>
  );
}

// export const mapDispatchToProps = dispatch => ({
//   addSearchResults = results => dispatch()
// });

export default VenueSearchBar;
