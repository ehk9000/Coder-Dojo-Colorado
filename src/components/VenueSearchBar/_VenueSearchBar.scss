.VenueSearchBar {
  text-align: left;
  
  h3 {
    font-size: 2rem;
    color: #ffffff;
    font-weight: 400;
  }

  p {
    color: #ffffff;
    margin: 5px 0;
  }

  input,
  button {
    border: none;
    margin: 30px 15px 0 0;
    height: 2.8rem;
    font-size: 1rem;
  }

  button {
    width: 150px;
    background-color: #298244;
    border: 2px solid #298244;
    color: #ffffff;
    border: none;
    font-weight: 300;

    &:hover {
      background: #144C25;
    }
  }

  input {
    width: 300px;
  }
}