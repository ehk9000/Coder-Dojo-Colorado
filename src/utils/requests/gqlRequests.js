import { gql } from 'apollo-boost';

export const GET_ALL_USERS = gql`
  {
    allUsers {
      id
      name
      username
      role
      guardianId{
        name
      }
      students{
        name
      }
    }
  }
`;

export const GET_USER_BY_TOKEN = gql`
  {
    me{
        username
        name
        role
    }
  }
`;

export const CREATE_STUDENT = gql`
  mutation createStudent(
    $username: String!,
    $password: String!,
    $name: String!,
    $birthdate: String!    
    ) {
    createStudent(
      username: $username,
      password: $password,
      name: $name,
      birthdate: $birthdate
    ) {
      id
    }
  }
`;

export const CREATE_GUARDIAN = gql`
  mutation createUser(
    $email: String!,
    $username: String!,
    $password: String!,
    $name: String!,
    $phoneNumber: String!,
    $street1: String!,
    $street2: String,
    $city: String!,
    $state: String!,
    $zip: String!
    ) {
    createUser(
      email: $email,
      username: $username,
      password: $password,
      name: $name,
      phoneNumber: $phoneNumber,
      street1: $street1,
      street2: $street2,
      city: $city,
      state: $state,
      zip: $zip
    ) {
      id
    }
  }
`;

export const CREATE_NEW_EVENT = gql`
  mutation createEvent(
    $name: String!,
    $venueId: Int!,
    $notes: String,
    $startTime: String!,
    $endTime: String!
    ) {
    createEvent(
      name: $name,
      venueId: $venueId,
      notes: $notes,
      startTime: $startTime,
      endTime: $endTime
    ) {
      id
    }
  }
`;

export const SIGN_IN = gql`
  mutation signIn(
    $username: String!,
    $password: String!    
    ) {
    signIn(
      username: $username,
      password: $password
    ) {
      token
      user{
        id
        username
        name
        role
      }
    }
  }
`;

export const GET_ALL_VENUES = gql`
  {
    allVenues{
      id
      name
      notes
      addresses{
        city
        street1
        street2
        zip
        state
        id
      }
    }
  }
`;

export const GET_VENUES_BY_CITY = gql`
  query allVenues($city: String!) {
    allVenues(city: $city) {
      id
      name
      notes
      addresses{
        city
        street1
        street2
        zip
        state
        id
      }
    }
  }
`;

export const GET_VENUE_DETAILS = gql`
  query allVenues($venueId: Int!) {
    allVenues(id: $venueId) {
      name
      email
      webUrl
      notes
      addresses {
        street1
        street2
        city
        state
        zip
      }
    }
  }
`;

export const UPCOMING_EVENTS_BY_VENUE = gql`
  query futureEvents($venueId: Int!) {
    futureEvents(venueId: $venueId){
      id
      name
      notes
      startTime
      endTime
    }
  }
`;

export const RECENT_EVENTS_BY_VENUE = gql`
  query pastEvents($venueId: Int!) {
    pastEvents(venueId: $venueId){
      id
      name
      notes
      startTime
      endTime
    }
  }
`;

export const GET_FAMILY = gql`
  {
    me {
      students {
        name
        username
        birthdate
        id
      }
    }
  }
`;

export const GET_ADMIN_DATA = gql`
  {
    me {
      email
      phoneNumber
      username
      addresses{
          street1
          street2
          city
          state
          zip
      }
      venues {
        id
        name
        notes
        email
        webUrl
        events {
          id
          name
          notes
          startTime
          endTime
          eventCode
        }
      }
    }
  }
`;

export const GET_EVENTS_ATTENDED = gql`
 {
    me{
        eventsAttended{
            name
            startTime
            id
        }
    }
}
`;

export const LOG_ATTENDANCE = gql`
  mutation logAttendance( 
    $eventCode: String!
    ) {
    logAttendance(
      eventCode: $eventCode
    ) {
      id
    }
  }
`;

export const GET_GUARDIAN_DATA = gql`
  {
    me {
      email
      phoneNumber
      addresses {
        street1
        street2
        city
        state
        zip
      }
    }
  }
`;
