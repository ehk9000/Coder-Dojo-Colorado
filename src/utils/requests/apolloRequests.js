import ApolloClient from 'apollo-boost';
import store from '../../store';
import { setFetching, setError } from '../../actions';

export function dispatchError(error) {
  const { message } = error.graphQLErrors[0];

  store.dispatch(setError(message));
  store.dispatch(setFetching(false));
}

export function setFetchingAndError() {
  const state = store.getState();
  const { isFetching, error } = state;

  if (!isFetching) store.dispatch(setFetching(true));
  if (error.length) store.dispatch(setError(''));
}

export async function apolloQuery(query, variables) {
  const client = new ApolloClient({
    uri: 'https://cryptic-stream-22818.herokuapp.com/graphql',
    request: (operation) => {
      operation.setContext({
        headers: {
          authorization: localStorage.getItem('token') || ''
        }
      });
    }
  });

  setFetchingAndError();

  let result;

  try {
    result = await client.query({ query, variables });
    if (result) store.dispatch(setFetching(false));
  } catch (error) {
    dispatchError(error);
  }

  return result ? result.data : false;
}

export async function apolloMutate(mutation, variables) {
  const client = new ApolloClient({
    uri: 'https://cryptic-stream-22818.herokuapp.com/graphql',
    request: (operation) => {
      operation.setContext({
        headers: {
          authorization: localStorage.getItem('token') || ''
        }
      });
    }
  });

  setFetchingAndError();

  let result;

  try {
    result = await client.mutate({ mutation, variables });
    if (result) store.dispatch(setFetching(false));
  } catch (error) {
    dispatchError(error);
  }
  
  return result ? result.data : false;
}
