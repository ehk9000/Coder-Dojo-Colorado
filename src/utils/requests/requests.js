import { apolloQuery, apolloMutate } from './apolloRequests';
import * as gql from './gqlRequests';

const requests = {
  signIn: user => apolloMutate(gql.SIGN_IN, user),
  createStudent: student => apolloMutate(gql.CREATE_STUDENT, student),
  createGuardian: guardian => apolloMutate(gql.CREATE_GUARDIAN, guardian),
  getAllUsers: () => apolloQuery(gql.GET_ALL_USERS),
  getUserByToken: () => apolloQuery(gql.GET_USER_BY_TOKEN),
  getAllVenues: () => apolloQuery(gql.GET_ALL_VENUES),
  getVenuesByCity: city => apolloQuery(gql.GET_VENUES_BY_CITY, city),
  getVenueDetails: venueId => apolloQuery(gql.GET_VENUE_DETAILS, venueId),
  getUpcomingEvents: venueId => apolloQuery(gql.UPCOMING_EVENTS_BY_VENUE, venueId),
  getRecentEvents: venueId => apolloQuery(gql.RECENT_EVENTS_BY_VENUE, venueId),
  getAdminDetails: () => apolloQuery(gql.GET_ADMIN_VENUES)
};

export default requests;
