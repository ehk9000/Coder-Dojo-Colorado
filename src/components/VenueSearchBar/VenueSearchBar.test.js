import React from 'react';
import { shallow } from 'enzyme';
import { VenueSearchBar, mapDispatchToProps } from './VenueSearchBar';
import * as actions from '../../actions';
import requests from '../../utils/requests/requests';

jest.mock('../../actions');

describe('VenueSearchBar', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<VenueSearchBar />);
  });

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should dispatch setSearchResults', () => {
    const searchResults = [{ name: 'Dojo' }];
    const action = actions.setSearchResults(searchResults);
    const dispatch = jest.fn();

    mapDispatchToProps(dispatch).setSearchResults(searchResults);

    expect(dispatch).toHaveBeenCalledWith(action);
  });
});
