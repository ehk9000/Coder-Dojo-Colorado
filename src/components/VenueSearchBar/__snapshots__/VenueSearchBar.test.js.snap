// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VenueSearchBar should match snapshot 1`] = `
<form
  className="VenueSearchBar"
  onSubmit={[Function]}
>
  <h3>
    Find a Dojo to Attend
  </h3>
  <p>
    Learn technology in an informal, creative and social environment. Find a dojo near you.
  </p>
  <label
    htmlFor="city-input"
  >
    <input
      id="city-input"
      onChange={[Function]}
      placeholder="Enter a city"
      type="text"
    />
  </label>
  <button
    className="city-search-btn"
    type="submit"
  >
    Find a Dojo
  </button>
</form>
`;
